frontend/src/components/TaskCard.tsx

```diff
@@ -19,22 +23,18 @@ const TaskCard: React.FC<TaskCardProps> = ({
     showCheckbox = false,
     task,
 }) => {
-
-const tagsArray = task.tags ? task.tags.split(" ").map(tag => tag.replace(/_/g, ' ')) : [];
     return (
         <div className="relative bg-white rounded-lg shadow-md p-6 mb-4">
             {showCheckbox && (
                 <input type="checkbox" className="relative top-2 w-6 h-6 mb-4" />
             )}
             {showCheckbox && (<hr />)}
             <MarkDown content={task.description} />
-            {task.linkTitle && (<h5 className="text-s pb-4">{task.linkTitle}</h5>)}
+            {task['link-title'] && (<h5 className="text-s pb-4">{task['link-title']}</h5>)}
             <MarkDown content={task.links} />
-            {tagsArray.map((tag, index) => (
-                <ResourceTagButton key={`${task.id}-${tag}`}>{tag}</ResourceTagButton>
-            ))}
+            <ResourceTagButton>{task.tag}</ResourceTagButton>
```

Comment:
Material UI Chip component should be used. ResourceTagButton can be removed.
https://mui.com/material-ui/react-chip/